src/extension.ts

```diff
@@ -508,6 +517,52 @@ class Device extends vscode.TreeItem {
             [this.shellPort.toString()]);
         term.show();
     }
+    
+    async captureScreenshot() {
+        const statusBarMessage = vscode.window.createStatusBarItem();
+        statusBarMessage.text = "Attempting to capture screenshot...";
+        statusBarMessage.show();
+
+        let anyErrors = false;
+        const handleCaptureError = (e) => {
+            anyErrors = true;
+            vscode.window.showErrorMessage("Error capturing screenshot: " + (e.message || e));
+            statusBarMessage.dispose();
+        }
+
+        try {
+            const screenshotDirectory = await getSharedTempDir('ev3dev-screenshots');
+            const screenshotBaseName = `ev3dev-${sanitizedDateString()}.png`;
+            const screenshotFile = `${screenshotDirectory}/${screenshotBaseName}`;
+
+            const conn = await this.exec('fbgrab -');
+            const writeStream = fs.createWriteStream(screenshotFile);
+
+            conn.on('error', (e: Error) => handleCaptureError);
+
+            writeStream.on('open', () => {
+                conn.stdout.pipe(writeStream);
+            });
+
+            writeStream.on('close', () => {
+                if (!anyErrors) {
+                    statusBarMessage.text = `Screenshot "${screenshotBaseName}" successfully captured`;
+                    vscode.commands.executeCommand('vscode.open', vscode.Uri.file(screenshotFile));

```

Comment:
Huh, good idea... I could probably even get fancier and check for the initial bytes of the PNG header (IIRC it literally starts with the ASCII text `PNG`). If it isn't a PNG, a warning could be displayed suggesting you may need to upgrade `fbcat`.